test/indexer/test_hca_indexer.py

```diff
@@ -597,8 +598,8 @@ def test_indexing_with_skipped_matrix_file(self):
                     file_names.add(file_name)
         self.assertEqual(4, len(entities_with_matrix_files))  # a project, a specimen, a cell suspension and a bundle
         self.assertEqual(aggregate_file_names, file_names)
```

Comment:
@jessebrennan I think i have to change this assertion here, if we are changing the name of the file within the indexer, then there will be a miss-match between the  list of `aggregations` and the list of `file_names`. I'm not sure how  this would be best handled;
1. Changing the value of the file.name as the contribution is being handled by the indexer.aggregation
1. Changing the assertion (although there might be tests in the future that have to be changed to follow suit)

I was leaning toward (1), however it seems a bit strange to be modifying an aggregation.

Reply:
Can you explain more about the miss-match? Which one doesn't have `/`s substituted? why haven't they been substituted?

Reply:
its the aggregations that have the miss-match; from the test:
```
{'377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0.zarr!.zattrs',
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0.zarr/.zattrs',
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_QCs.csv',
 ...
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_rsem.isoforms.results',
 'R1.fastq.gz',
 'R2.fastq.gz'} != 
{'377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0.zarr/.zattrs',
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_QCs.csv',
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_bait_bias_detail_metrics.csv',
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_base_distribution_by_cycle_metrics.csv',
 ...
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_rsem.genes.results',
 '377f2f5a-4a45-4c62-8fb0-db9ef33f5cf0_rsem.isoforms.results',
 'R1.fastq.gz',
 'R2.fastq.gz'}
```
I'm not sure why the aggregation has both the original name and the modified name, ill have to dig in deeper into that section of the code base to understand it more.